grpcx/server.go

```diff
@@ -119,3 +123,37 @@ func WithDiscardedCodes(codes ...codes.Code) func(*errorLogsOptions) {
 		logsConfig.discardedCodes = append(logsConfig.discardedCodes, codes...)
 	}
 }
+
+type recoveryOptions struct {
+	exitFunc func()
+}
+
+func NewRecoveryHandler(l logx.Logger, options ...func(*recoveryOptions)) func() {
+	ro := &recoveryOptions{
+		exitFunc: func() { os.Exit(2) },
+	}
+
+	for _, option := range options {
+		option(ro)
+	}
+
+	return func() {
+		if r := recover(); r != nil {
+			l.Info(fmt.Sprintf("%v", r), logx.F("stack_trace", strings.Split(string(debug.Stack()), "\n")))
+			ro.exitFunc()
+		}
+	}
+}
+
+func SetExitFunction(exitFunc func()) func(*recoveryOptions) {
+	return func(ro *recoveryOptions) {
+		ro.exitFunc = exitFunc
+	}
+}
```

Comment:
the functions above are not bounded to grpcx, so they should be in another package.